### PR TITLE
CRM-20707 include start/end dates when updating membership

### PR DESCRIPTION
**Pull request by lcdservices**
* [CRM-20707: membership inheritance does not update start\/end dates](https://issues.civicrm.org/jira/browse/CRM-20707)

### CODE REVIEW

File: CRM/Member/BAO/Membership.php [hunk 1]
```diff
@@ -1464,6 +1464,10 @@ public static function createRelatedMemberships(&$params, &$dao, $reset = FALSE)
         unset($params['lineItems']);
         unset($params['line_item']);
 
+        //CRM-2707 - include start/end date
```
Comment (mlutfy) [inline CRM/Member/BAO/Membership.php]: Typo in comment, should be CRM-20707 ?

**Comment by jackrabbithanna** [comment]
@lcdservices can you reroll this PR to resolve the conflicts?

**Comment by jackrabbithanna** [comment]
I am seeing this condition on one of my client sites.

In my scenario, we have organization contacts that hold the parent membership. Individuals can then inherit the memberships by being employees of the organization.

They also have "branches" which are organization contacts themselves, and can be related to the "parent" organizations

Other individuals can be employees of the branches, and these employees also inherit the membership from the parent organization, through the relationship to the branch.

What I am seeing is that when a parent membership is renewed, the inherited memberships will update fine UNTIL one of the branch organizations and its employees are processed...

The next employee of the parent organization that is processed will not get their inherited membership dates updated, and none of the rest will afterwards.

This PR appears to resolve the issue for me

**Comment by jackrabbithanna** [comment]
I'm testing on 4.7.27

**Comment by jackrabbithanna** [comment]
@eileenmcnaughton  or @seamuslee001 do you'll have any thoughts on this?

**Comment by seamuslee001** [comment]
I think this makes sense but i do note that there is a conflict on the file now which may have some issues. @lcdservices Brian are you able to rebase this against the current master?

**Comment by lcdservices** [comment]
@seamuslee001 @jackrabbithanna I've rebased the PR. Trivial conflict.

**Comment by eileenmcnaughton** [comment]
I normally would not merge something like this without a unit test. However, this has been stalled in the review queue for too long for me to feel ok requiring that. Can you please squash to one commit though?

**Comment by lcdservices** [comment]
@eileenmcnaughton it's now a single commit

**Comment by colemanw** [comment]
@civicrm-builder retest this please

**Pull request merged**